Zend/zend_alloc.c

```diff
@@ -3036,10 +3023,13 @@ ZEND_API bool zend_mm_is_custom_heap(zend_mm_heap *new_heap)
 #endif
 }
 
-ZEND_API bool zend_mm_is_observed(zend_mm_heap *new_heap)
+ZEND_API bool zend_mm_is_observed(zend_mm_heap *heap)
 {
 #if ZEND_MM_CUSTOM
-	return (AG(mm_heap)->use_custom_heap & ZEND_MM_CUSTOM_HEAP_OBSERVED) != 0;
+	if (!heap && !(heap = AG(mm_heap))) {
+		return false;
+	}
```

Comment:
This is probably a bug if both heap and AG(mm_heap) are NULL. We should assume it's not the case, and add a ZEND_ASSERT (here and in other functions bellow) :

```
if (!heap) {
    heap = AG(mm_heap);
}
ZEND_ASSERT(heap);
```

(We should also avoid assignments in conditions, as this is error prone and harder to read)